### PR TITLE
react-markdown: Upgrade to version 8.0.7 to be compatible with rollup

### PR DESCRIPTION
**Pull request by YextSLu**
J=CLIP-1321
TEST=manual|auto

Ran `npm run test` and `npm run build` to ensure no errors and manually interact with `test-site` to verify basic functionality

### CODE REVIEW

File: src/components/Markdown.tsx [hunk 1]
```diff
@@ -1,7 +1,5 @@
-import ReactMarkdown, {
-  PluggableList,
-  ReactMarkdownOptions,
-} from "react-markdown";
+import ReactMarkdown, { Options } from "react-markdown";
+import { PluggableList } from "unified";
```
Comment (yen-tt) [inline src/components/Markdown.tsx:2]: lets avoid using types directly from a transitive dependency. Can you update L11 to use `Options` from react-markdown instead?
`{ remark?: Options['remarkPlugins']; rehype: Options['rehypePlugins'] }`

File: test-site/package.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
     "@yext/chat-ui-react": "file:..",
     "react": "file:../node_modules/react",
     "react-dom": "file:../node_modules/react-dom",
+    "react-markdown": "^8.0.7",
```
Comment (yen-tt) [inline test-site/package.json:10]: this should not be necessary

File: jest.config.json [hunk 3]
```diff
@@ -12,7 +12,8 @@
   "clearMocks": true,
   "testMatch": ["<rootDir>/tests/**/*.(test).ts(x)?"],
   "moduleNameMapper": {
-    "\\.(css|scss|less)$": "<rootDir>/tests/__utils__/style-mock.ts"
+    "\\.(css|scss|less)$": "<rootDir>/tests/__utils__/style-mock.ts",
+    "react-markdown": "<rootDir>/node_modules/react-markdown/react-markdown.min.js"
```
Comment (yen-tt) [inline jest.config.json:16]: can you make a comment here why this is needed

File: test-site/package.json (hunk 2 above)
Comment (YextSLu) [reply test-site/package.json]: Removed

File: src/components/Markdown.tsx (hunk 1 above)
Comment (YextSLu) [reply src/components/Markdown.tsx]: Fixed

File: jest.config.json (hunk 3 above)
Comment (yen-tt) [reply jest.config.json:16]: what are the syntax issues? I tested and build from your branch with just the `moduleNameMapper` change and didn't encounter any

Comment (YextSLu) [reply jest.config.json:16]: I should note that keeping either the`moduleNameMapper` line for `react-markdown`, or using the `transformIgnorePatterns` line works. Both are not needed

Screenshot 2024-07-08 at 12 07 26 PM

Comment (yen-tt) [reply jest.config.json:16]: to clarify, my original comment is asking if we can add a comment in the config file

Comment (YextSLu) [reply jest.config.json:16]: Added

File: jest.config.json [hunk 4]
```diff
@@ -12,7 +12,9 @@
   "clearMocks": true,
   "testMatch": ["<rootDir>/tests/**/*.(test).ts(x)?"],
   "moduleNameMapper": {
-    "\\.(css|scss|less)$": "<rootDir>/tests/__utils__/style-mock.ts"
+    "\\.(css|scss|less)$": "<rootDir>/tests/__utils__/style-mock.ts",
+    "$comment": "react-markdown throws syntax errors (ESM-related). This imports the .js file directory without issue",
```
Comment (yen-tt) [inline jest.config.json:16]: the comment shouldn't be an entry in `moduleNameMapper`. Can you update it to be:
`\\react-markdown throws syntax err...`

**Review by yen-tt — APPROVED** [review]

**Pull request merged**